templates/common/_base/files/configure-ovs-network.yaml

```diff
@@ -341,58 +288,136 @@ contents:
       counter=0
       while [ $counter -lt 5 ]; do
-        if nmcli conn up ovs-if-br-ex; then
+        if nmcli conn up "$ovs_interface"; then
           echo "OVS successfully configured"
+          update_nm_conn_files "$bridge_name" "$port_name"
           copy_nm_conn_files
-          ip a show br-ex
+          ip a show "$bridge_name"
           ip route show
           nmcli c show
           configure_driver_options ${iface}
-          exit 0
+          configured=true
+          break
         fi
         sleep 5
         counter=$((counter+1))
       done
+      if [ $configured != true ]; then
+        echo "ERROR: Failed to activate $ovs_interface NM connection"
+        exit 1
+      fi
+    }
 
-      echo "ERROR: Failed to activate ovs-if-br-ex NM connection"
+    if ! rpm -qa | grep -q openvswitch; then
+      echo "Warning: Openvswitch package is not installed!"
       exit 1
+    fi
+    echo "Current routing and connection state:"
+    ip route show
+    nmcli c show
+    if [ "$1" == "OVNKubernetes" ]; then
+      # Configures NICs onto OVS bridge "br-ex"
+      # Configuration is either auto-detected or provided through a config file written already in Network Manager
+      # key files under /etc/NetworkManager/system-connections/
+      # Managing key files is outside of the scope of this script
+      # if the interface is of type vmxnet3 add multicast capability for that driver
+      # REMOVEME: Once BZ:1854355 is fixed, this needs to get removed.
+      extra_bridge_file='/etc/ovnk/extra_bridge'
+      function configure_driver_options {
+        intf=$1
+        driver=$(cat "/sys/class/net/${intf}/device/uevent" | grep DRIVER | awk -F "=" '{print $2}')
+        echo "Driver name is" $driver
+        if [ "$driver" = "vmxnet3" ]; then
+          ifconfig "$intf" allmulti
+        fi
+      }
+      iface=""
+      counter=0
+      # find default interface
+      while [ $counter -lt 12 ]; do
+        # check ipv4
+        iface=$(ip route show default | awk '{ if ($4 == "dev") { print $5; exit } }')
+        if [[ -n "$iface" ]]; then
+          echo "IPv4 Default gateway interface found: ${iface}"
+          break
+        fi
+        # check ipv6
+        iface=$(ip -6 route show default | awk '{ if ($4 == "dev") { print $5; exit } }')
+        if [[ -n "$iface" ]]; then
+          echo "IPv6 Default gateway interface found: ${iface}"
+          break
+        fi
+        counter=$((counter+1))
+        echo "No default route found on attempt: ${counter}"
+        sleep 5
+      done
+      if [ "$iface" != "br-ex" ] && [  -f "$extra_bridge_file" ] && [ -d "/sys/class/net/br-ex" ]; then
+        echo "default gateway is not bridge but bridge exists, reverting"
+        update_nm_conn_files br-ex phys0
+        rm_nm_conn_files
+        nmcli c reload
+        sleep 5
+      fi
+
+      convert_to_bridge "$iface" "br-ex" "phys0"
+
+      if [ -f "$extra_bridge_file" ] && [ ! -d "/sys/class/net/br-ex1" ]; then
```

Comment:
so you cant check default gateway to ensure the bridge is configure correctly so you are using /sys/class iiuc. Would it be better to use nmcli conn show --active and see if br-ex1 and its phys iface are up?

Reply:
Yes will do